### PR TITLE
Chore/126 refactor config manager

### PR DESCRIPTION
**Pull request by phmngocnghia**
update #126

### CODE REVIEW

File: packages/aml-checklist/tsconfig.json [hunk 1]
```diff
@@ -10,5 +10,5 @@
       "logger": ["../../scripts/logger/sentry-logger.ts"]
     }
   },
-  "include": ["src"]
+  "include": ["src", "../../reapt-config.d.ts"]
```
Comment (willmcvay) [inline packages/aml-checklist/tsconfig.json]: Typo, reapit spelt wrong

File: packages/config-manager/src/utils/generateConfigTsDef.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+const { json2ts } = require('json-ts')
```
Comment (willmcvay) [inline packages/config-manager/src/utils/generateConfigTsDef.js]: File name kebab-case please

File: packages/config-manager/src/utils/handleFetchingSecretFail.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+const { writeConfigInCWD } = require('./writeConfigInCWD')
```
Comment (willmcvay) [inline packages/config-manager/src/utils/handleFetchingSecretFail.js]: File name again please

File: packages/config-manager/src/utils/writeConfigInCWD.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+const fs = require('fs')
```
Comment (willmcvay) [inline packages/config-manager/src/utils/writeConfigInCWD.js]: And again

File: packages/geo-diary/tsconfig.json [hunk 5]
```diff
@@ -10,5 +10,5 @@
       "logger": ["../../scripts/logger/sentry-logger.ts"]
     }
   },
-  "include": ["src"]
+  "include": ["src", "../../reapt-config.d.ts"]
```
Comment (willmcvay) [inline packages/geo-diary/tsconfig.json]: Nice

File: packages/config-manager/src/index.js [hunk 6]
```diff
@@ -12,50 +17,120 @@ const createSecret = secretName => {
...
+    const data = await secretsManager
+      .updateSecret({
+        SecretId: secretName,
+        SecretString: JSON.stringify(updatedConfig),
+      })
+      .promise()
+
+    return console.log('Successfully updated secret', data)
+  } catch (err) {
+    console.log('Something went wrong when updating configuration. Detailed error with stack trace is provided below:')
```
Comment (tanphamhaiduong) [inline packages/config-manager/src/index.js]: console.error

File: packages/config-manager/src/utils/generateConfigTsDef.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+      rootName: 'ProcessEnv',
+      prefix: '',
+    })
+
+    let formattedGeneratedTsDef = `export ${formatCode(generatedTsDef)}`
+    fs.writeFileSync(REAPIT_TS_DEF_IN_CWD_PATH, formattedGeneratedTsDef)
+
+    console.log(`${REAPIT_TS_DEF_NAME} has been saved - be sure to add to your .gitignore file`)
+  } catch (err) {
+    console.log(
```
Comment (tanphamhaiduong) [inline packages/config-manager/src/utils/generateConfigTsDef.js]: console.error

File: packages/config-manager/src/utils/generateConfigTsDef.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+      'Something went wrong when generating typescript definition. Detailed error with stack trace is provided below:',
+    )
+    console.log(err, err.stack)
```
Comment (tanphamhaiduong) [inline packages/config-manager/src/utils/generateConfigTsDef.js]: console.error

File: packages/config-manager/src/utils/handleFetchingSecretFail.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+module.exports = (err, isGenerateConfigTsDef) => {
+  console.log('Something went wrong when fetching config. Detailed error with stack trace is provided below:')
+  console.error(err, err.stack)
+  console.log('A base configuration file will be provided')
+
+  let config = ''
+  try {
+    config = fs.readFileSync(REAPIT_BASE_CONFIG_PATH)
+  } catch (err) {
+    console.log(
```
Comment (tanphamhaiduong) [inline packages/config-manager/src/utils/handleFetchingSecretFail.js]: console.error

**Pull request merged**